server/api/collections.js

```diff
@@ -59,14 +86,17 @@ router.post("collections.create", auth(), async (ctx) => {
     private: isPrivate,
     sharing,
     sort,
+    index,
   });
 
+  const collectionsIdWithIndex = await removeIndexCollisions(user.teamId);
```

Comment:
The use of this method is pretty much the only thing I'm left questioning in this PR, it feels very brute force. We already know the new `index` we're saving here, why don't we just check if it exists in the team already with a specific query rather than querying all collections in the team..

Reply:
Actually, this is how it should have been called

```js
  const collectionsIdWithIndex = await removeIndexCollisions(
    user.teamId,
    index,
    collections
  );
```
But again, all this logic was for my previous implementation. Earlier, I thought to remove the index collision of not only the `new index` but other indexes as well. Now that I think of it I unable to create a specific scenario where this is necessary. 

I will push your suggestion in an hour if I cannot contrive an example.